include/exadg/incompressible_navier_stokes/user_interface/parameters.cpp

```diff
@@ -1113,9 +1109,7 @@ Parameters::print_parameters_momentum_step(dealii::ConditionalOStream const & pc
 
   if(update_preconditioner_momentum == true)
   {
-    // if a nonlinear problem has to be solved
-    if(equation_type == EquationType::NavierStokes and
-       treatment_of_convective_term == TreatmentOfConvectiveTerm::Implicit)
+    if(nonlinear_problem_has_to_be_solved())
```

Comment:
@richardschu It seems to me that we forgot to update these if-statements in the past, e.g. when introducing a nonlinear viscous problem. Can you confirm this and do you agree to the new code version?

Reply:
Yes I/we forgot about this part here!
Thanks for catching it now :+1: